index.js

```diff
@@ -1,79 +1,157 @@
-if (shouldShowResultInDogYears) {
-  console.log("Your dog will be " + ageInDogYears + " dog years old in " + dogYearFuture + ".");
-} else {
-  console.log("Your dog will be " + dogYear + " human years old in " + dogYearFuture + ".");
+function getFullname(firstname,surname){
+  return firstname + " " + surname;
+ 
 }
-//task 3
-// Function to calculate house price
-function calculateHousePrice(width, depth, height, gardenSize) {
-    const volumeInMeters = width * depth * height;
-    const housePrice = volumeInMeters * 2.5 * 1000 + gardenSize * 300;
-    return housePrice;
-  }
-  
-  // Peter's house details
-  const peterWidth = 8;
-  const peterDepth = 10;
-  const peterHeight = 10;
-  const peterGardenSize = 100;
-  const peterActualCost = 2500000;
-  
-  const peterCalculatedPrice = calculateHousePrice(peterWidth, peterDepth, peterHeight, peterGardenSize);
-  console.log("Peter's calculated house price: " + peterCalculatedPrice);
-  console.log("Peter's actual house cost: " + peterActualCost);
-  if (peterCalculatedPrice > peterActualCost) {
-    console.log("Peter is paying less than the calculated price. Good deal!");
-  } else {
-    console.log("Peter is paying more than the calculated price. Not a good deal.");
-  }
-  
-  // Julia's house details
-  const juliaWidth = 5;
-  const juliaDepth = 11;
-  const juliaHeight = 8;
-  const juliaGardenSize = 70;
-  const juliaActualCost = 1000000;
-  
-  const juliaCalculatedPrice = calculateHousePrice(juliaWidth, juliaDepth, juliaHeight, juliaGardenSize);
-  console.log("Julia's calculated house price: " + juliaCalculatedPrice);
-  console.log("Julia's actual house cost: " + juliaActualCost);
-  if (juliaCalculatedPrice > juliaActualCost) {
-    console.log("Julia is paying less than the calculated price. Good deal!");
-  } else {
-    console.log("Julia is paying more than the calculated price. Not a good deal.");
+
+console.log(getFullname("Benjamin","Hughes"));
+
+let fullname1 = getFullname("Benjamin", "Hughes");
+let fullname2 = getFullname("Emma", "Smith");
+
+console.log(fullname1); // returns "Benjamin Hughes"
+console.log(fullname2); // returns "Emma Smith"
+
+
+//Second 
+function getFullname(firstname, surname, useFormalName = false, gender = "female") {
+  let title = "";
+
+  if (useFormalName) {
+      if (gender.toLowerCase() === "male") {
+          title = "Lord ";
+      } else {
+          title = "Lady ";
+      }
   }
-  //task 4
-// Step 1: Create the arrays
-const firstWords = ["Easy", "Awesome", "Corporate", "Creative", "Innovative", "Dynamic", "Smart", "NextGen", "Tech", "Green"];
-const secondWords = ["Solutions", "Enterprises", "Ventures", "Technologies", "Networks", "Systems", "Hub", "Labs", "Dynamics", "Corporation"];
 
-// Step 2: Generate random indices
-const randomIndex1 = Math.floor(Math.random() * 10);
-const randomIndex2 = Math.floor(Math.random() * 10);
+  return title + firstname + " " + surname;
```

Comment:
@melo4yxa77796 Great work on this function, it works as expected! I've got a little suggestion, to make this function even more bulletproof. 

Imagine if someone calls it like this `getFullname("Yuliia")`. We can add a default value for the second argument `getFullname(firstname, surname = "", ...)`, but sadly the result of the function will become `"Yuliia "` (with an extra whitespace at the end).

Let's try to adjust the function to handle this use case.

<details>
  <summary>Hint</summary>
  
  One of the options would be to introduce an extra variable called `fullname` (or `result`) and then concatenate it with the `surname` conditionally:

```js
let fullname = "";

// ... format title and first name

if (surname !== "") {
  fullname += surname;
}
``` 
</details>